fix(groups): Group membership request check existing requests

lowfill opened the pull request:
Added check of exisiting request and added error message when an existing request exists

Fixes: #8901

hypeJunction commented on mod/groups/actions/groups/membership/join.php:
```diff
@@ -44,35 +44,40 @@
 		}
 	} else {
-		add_entity_relationship($user->guid, 'membership_request', $group->guid);
+		if(!check_entity_relationship($user->guid,'membership_request',$group->guid)){
```
We prefer early termination:
```
if () {

} else if (check_entity_relationship()) {
   register_error();
   forward(REFERRER);
} else {

}
```

hypeJunction commented on mod/groups/languages/en.php:
```diff
@@ -134,6 +134,7 @@
 	'groups:joinrequestmade' => 'Requested to join group',
 	'groups:joined' => 'Successfully joined group!',
 	'groups:left' => 'Successfully left group',
+	'groups:requestexists' => 'You alredy requested join this group, please wait the group owner to respond.',
```
`You have already requested to join this group`, should be enough.

mrclay replied:
`You already requested to join this group. Please wait for the group owner to respond.`

mrclay replied:
Agreed.

jdalsem replied:
typo in alredy..

mrclay commented on mod/groups/actions/groups/membership/join.php:
```diff
@@ -44,35 +44,40 @@
 			register_error(elgg_echo("groups:cantjoin"));
 		}
 	} else {
-		add_entity_relationship($user->guid, 'membership_request', $group->guid);
```
Instead use a return early strategy, which (in addition to other benefits) makes the diff cleaner to read. E.g.

```php
if (check_entity_relationship($user->guid, 'membership_request', $group->guid)) {
	register_error(elgg_echo("groups:requestexists"));
	forward(REFERER);
}
```

Also we enforce a [code style](http://learn.elgg.org/en/stable/contribute/code.html#php-guidelines) close to PSR-2. The Travis build will tell you if you get it wrong.

mrclay commented:
Oops, 2 reviews saying same thing :)

mrclay commented on mod/groups/languages/en.php:
```diff
@@ -134,6 +134,7 @@
 	'groups:joinrequestmade' => 'Requested to join group',
 	'groups:joined' => 'Successfully joined group!',
 	'groups:left' => 'Successfully left group',
+	'groups:requestexists' => 'You alredy requested join this group.',
```
`You already requested to join this group.`

mrclay commented on mod/groups/actions/groups/membership/join.php:
```diff
@@ -43,7 +43,11 @@
 			register_error(elgg_echo("groups:cantjoin"));
 		}
-	} else {
+	} else if(check_entity_relationship($user->guid,'membership_request',$group->guid)) {
```
- space after `if` (not a function call)
- space after each `,` in argument list
- `} else {`, not Drupal style

lowfill commented:
I don't understand was Travis is saying

mrclay commented:
See https://gitter.im/Elgg/Elgg?at=57e67afe00df3cc4224173dc

lowfill committed:
fix(groups): Group membership request check existing requests

Added check of exisiting request and added error message when an existing request exists

Fixes: #8901

mrclay commented on mod/groups/languages/en.php:
```diff
@@ -134,6 +134,7 @@
 	'groups:joinrequestmade' => 'Requested to join group',
 	'groups:joined' => 'Successfully joined group!',
 	'groups:left' => 'Successfully left group',
+	'groups:requestexists' => 'You already requested join this group.',
```
"requested **to** join"

hypeJunction replied:
I think proper English is `You have already requested` if no time is specified. Past tense is used if you specify time `You requested to joint his group 2 days ago`.

hypeJunction commented:
@lowfill Thanks for your contribution. I have used it as a basis for a different pull request here: #10379 10379
